packages/test/test-end-to-end-tests/src/test/localTestSignals.spec.ts

```diff
@@ -31,6 +49,29 @@ const waitForSignal = async (...signallers: { once(e: "signal", l: () => void):
 		),
 	);
 
+const waitForTargetedSignal = async (
+	targetedSignaller: { once(e: "signal", l: () => void): void },
+	otherSignallers: { once(e: "signal", l: () => void): void }[],
+) =>
+	Promise.all([
+		timeoutPromise((resolve) => targetedSignaller.once("signal", () => resolve()), {
+			durationMs: 2000,
+			errorMsg: `Targeted Signaller Timeout`,
+		}),
+		otherSignallers.map(async (signaller, index) =>
+			timeoutPromise(
+				(reject) =>
```

Comment:
re: "This doesn't look like resolve, it looks like reject." -- it is meant to be `reject`, the otherSignallers are the non-targeted clients that should not receive a signal.

Reply:
Would you recommend to remove usage of `timeoutPromise` function from the tests? I see we use it in the `waitForSignal` function, so I assumed it is safe to use.

Reply:
No. It can be used safely; its signature just makes it easy to use incorrectly. If the signature were different, it would be hard to get wrong. We should update the signature, if it doesn't make some patterns too hard or impossible. We'd have to try out the change and see.

Reply:
Ok so something like this be better?: 
```typescript
export async function timeoutPromise<T = void>(
    executor: (
        controller: {
            resolve: (value: T | PromiseLike<T>) => void,
            reject: (reason?: any) => void,
        }
    ) => void,
): Promise<T>
```
It is used pretty extensively in our end to end tests but I could work on switching them if this is the safer pattern to use.

Reply:
Exactly. And it can be after.

Reply:
Great. Created a work item in ADO for this here I'll start to work on: https://dev.azure.com/fluidframework/internal/_workitems/edit/8235

Reply:
You still need to fix this case, though. ;)

Reply:
I think I'm misunderstanding what's meant to fix here in this particular case. Maybe I can explain what I'm trying to do with this function so I can understand the problem. Instead of the default TImeoutWithError, I specify to TimeoutWithValue in the `timeoutOptions` so that when the `otherSignallers` timeouts after not receiving a signal timeoutPromise resolves with a specific value rather than rejecting the promise. https://github.com/microsoft/FluidFramework/blob/3d83ec9e7d07fef1b7cdb4cc5863ae9998f9642e/packages/test/test-utils/src/timeoutUtils.ts#L163C2-L174C16

Reply:
The lambda you have on line 66 has a parameter named `reject`, but that parameter is actually the `resolve` parameter. `reject` is the second parameter.
It would not hurt to create the revise `timeoutPromise` in this file to make all of the uses clear. You can `import { timeoutPromise as timeoutPromiseUnnamed, ...` and call `timeoutPromiseUnnamed` to implement `timeoutPromise` with the improved signature.
Or maybe I have something wrong, but I don't think I do.

Reply:
Ok sorry I understand now -- made this fix here.